test/fakeintake/server/server_test.go

```diff
@@ -244,6 +243,25 @@ func TestServer(t *testing.T) {
 		fi.handleFlushPayloads(response, request)
 		assert.Equal(t, http.StatusAccepted, response.Code, "unexpected code")
 	})
+
+	t.Run("should clean payloads older than 15 minutes", func(t *testing.T) {
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do look for something to sincerely praise. -->
**👏 praise**
Nice for the tes, can you add one where you ensure that if you have one old and one fresh payload, you keep the fresh one only ?

Reply:
Done!